src/libstore/derivations.cc

```diff
@@ -154,31 +155,76 @@ StorePath writeDerivation(Store & store,
 }
 
 
+namespace {
+/* This mimics std::istream to some extent. We use this much smaller implementation
+   instead of plain istreams because the sentry object overhead is too high. */
+struct ParseContext {
+    std::string_view str;
+
+    int peek() const {
+        return str.empty() ? EOF : str[0];
+    }
+
+    int get() {
+        if (str.empty()) return EOF;
+        char c = str[0];
+        str.remove_prefix(1);
+        return c;
+    }
+};
+
+constexpr struct Escapes {
+    char map[256];
+    constexpr Escapes() {
+        for (int i = 0; i < 256; i++) map[i] = (char) (unsigned char) i;
+        map[(int) (unsigned char) 'n'] = '\n';
+        map[(int) (unsigned char) 'r'] = '\r';
+        map[(int) (unsigned char) 't'] = '\t';
+    }
+    char operator[](char c) const { return map[(unsigned char) c]; }
+} escapes;
+}
+
+
 /* Read string `s' from stream `str'. */
-static void expect(std::istream & str, std::string_view s)
+static void expect(ParseContext & str, std::string_view s)
 {
-    for (auto & c : s) {
-        if (str.get() != c)
-            throw FormatError("expected string '%1%'", s);
-    }
+    if (!str.str.starts_with(s))
+        throw FormatError("expected string '%1%'", s);
+    str.str.remove_prefix(s.size());
 }
 
 
 /* Read a C-style string from stream `str'. */
-static std::string parseString(std::istream & str)
+static BackedStringView parseString(ParseContext & str)
 {
-    std::string res;
     expect(str, "\"");
-    int c;
-    while ((c = str.get()) != '"')
-        if (c == '\\') {
-            c = str.get();
```

Comment:
resolved:
Looks like we had a "bug" where we accepted invalid strings and inserted `\xFF`, but only when parsing a bare ATerm string (ie not in a tuple or anything), which I don't think ever occurred.

Reply:
Worth a release note?

Reply:
I think it is worth noting. Thanks!